### ISSUE TITLE
Right click item -> Duplicate Entry

### ISSUE THREAD

**Issue by Jamonnin3** [geheur/bank-tag-custom-layouts#11]
It would be fantastic to be able to duplicate entries for items. My biggest gripe was not being able to just click through all the items and have the layout be perfect every time. Currently I need to click the angler twice in order to continue through my whole setup. It would be nice to be able to have duplicate entries to simply click through the whole inventory. This would be especially nice to remember exactly how many prayer potions to take on a task or similar things. Having up to 28 duplicates of any item would be great.

![banktaglayoutsduplication](https://user-images.githubusercontent.com/22843757/112742758-5c694500-8f5f-11eb-9230-cd6d79240c7c.png)

**Comment by KiwiZ0** [comment]
I would LOVE this. Would be the closest thing to just having presets like RS3 lol

**Comment by geheur** [comment]
Would also be nice for different sets of runes (e.g. having humidify in a separate column from vengeance despite sharing astral runes).

![image](https://user-images.githubusercontent.com/41499327/114494162-b8270580-9bd0-11eb-8f5b-cb9597327bb6.png)

I'd also use it for a "recharging items" tab.

![image](https://user-images.githubusercontent.com/41499327/114494368-1bb13300-9bd1-11eb-8f8a-4668bd60b9c6.png)

**Comment by geheur** [comment]
Auto-layout should be updated to use this when you have multiple of an item in your inventory.

Might also be nice to add an option to auto-layout to organize the equipped and inventory items in an aesthetically nice way like the picture in the OP. Slower to withdraw but looks pretty.

**Comment by KiwiZ0** [comment]
You could just arrange it so you can do a 2x1 down and it's all in place but yeah still dying for this feature

**Comment by geheur** [comment]
WidgetItemOverlays don't work too well with fake items :/

![image](https://user-images.githubusercontent.com/41499327/115131256-913f4980-9fab-11eb-8f28-9d977291ffab.png)

**Comment by Jamonnin3** [comment]
> 
> 
> WidgetItemOverlays don't work too well with fake items :/
> 
> ![image](https://user-images.githubusercontent.com/41499327/115131256-913f4980-9fab-11eb-8f28-9d977291ffab.png)
> 
> Only shows on one of the rune pouches because the other two are just graphics.
> 
> I could add fake widget items in but some plugins use the bank item container to determine the contents of your bank and this might confuse them.

Honestly I don't see this as a negative. It would very rarely be used and probably never matter. Imo it's a nonissue.

**Comment by KiwiZ0** [comment]
Wait was this added? I redownloaded it and I don't have this option

**Comment by geheur** [comment]
It has not been added unfortunately. It's mostly done but I haven't been in the right mood for finishing it off :(.

**Comment by KiwiZ0** [comment]
Oh damn, I'm sorry about that but if it gives you the motivation I've been DYING for this haha

**Comment by geheur** [comment]
Might have to walk back on "it's mostly done" since this becomes extremely nasty when you consider variant items (both those that are equivalent like different-mutagen serp helms and those which aren't, like different degradation states of barrows armor).

tl;dr more thought on edge cases and testing required.

For something like a serp helm, if you duplicate it, I would want to put that exact item id in all duplicate spots. But for barrows items, you can't do this because once something degrades from 100 -> 75 you'd want the 75 to show up in all duplicate spots. What about ironmen with one degraded ahrims top and 5 fully repaired ahrims tops? They'll want to withdraw the degraded one if it exists, but if it doesn't exist they'll want to withdraw the fully repaired version. And runelite bank tags considers all barrows items variant and will put them in the same tab. So the full ones will appear alongside the degraded duplicates.

I think the barrows item situation can be solved by having custom logic where the plugin just hides all barrows items that it thinks you don't want (i.e. it will put in all slots for that item the degraded version if it exists, otherwise the full version). This will however break tabs where people for whatever reason want to see all their barrows items. So it should probably be toggleable. But without this toggled on, barrows items will remain broken as shit (although that's exactly how it is right now so no big deal lol :D, except that if people try to duplicate it now it will probably break even harder and if people do things wrong they'd inadvertently create duplicate items - or can they? what happens if my plugin sees 2 ahrims tops 100? Cause for bank fillers it only shows them once and hides the dupes somehow).

What happens if you have duplicated regular serps and convert your serp to the mutated version?

Maybe an easy solution is: block duplicates on variants... who really needs to dupe a piece of gear? But potions are variant items! So this feature might require #13 to be resolved first.

A lot of these situations can be pretty safely ignored because they're so rare but I know stuff like barrows items currently works in bad/unintuitive ways and I feel like they should maybe be fixed at the same time?

**Comment by KiwiZ0** [comment]
PLS T.T

**Comment by ddeamaral** [comment]
This would be such a great feature! pleeaaaassssssseeee

**Comment by geheur** [comment]
![image](https://user-images.githubusercontent.com/41499327/120753357-e71a7300-c4bf-11eb-9710-4aa4c381f4c5.png)
haha I hope to open a PR tomorrow :) Just have to figure out how auto-layout is supposed to work with this.

**Comment by Jamonnin3** [comment]
> ![image](https://user-images.githubusercontent.com/41499327/120753357-e71a7300-c4bf-11eb-9710-4aa4c381f4c5.png)
> haha I hope to open a PR tomorrow :) Just have to figure out how auto-layout is supposed to work with this.
> 
> Sorry for the wait.

I would personally prefer it to automatically duplicate the items in the auto layout. If you don't want duplicates you can just put one of each item in your inventory when setting up the layout instead. But there is no alternative if you want duplicates.

**Comment by geheur** [comment]
I'm thinking a numerical limit. I.e. if there's more than 4 (customizable,
probably) of the item in a row, do not make duplicates.

**Comment by geheur** [comment]
PR open: https://github.com/runelite/plugin-hub/pull/1500
